### PR TITLE
reuse ssh terminal when configured

### PR DESCRIPTION
**Pull request by yungezz**
## Summary

Issue #119 #73 . Support reuse ssh terminal per configuraiton.

## Issue Type

- New Feature

### CODE REVIEW

File: src/sshRunner.ts [hunk 1]
```diff
@@ -19,12 +19,24 @@ const browseThePC = 'Browse the PC..';
...
+    private terminalList: { [key: string]: vscode.Terminal } = {};
 
     constructor(outputChannel: vscode.OutputChannel) {
         super(outputChannel);
 
         this.folderSyncer = new FolderSyncer(outputChannel);
 
+        vscode.window.onDidCloseTerminal((terminal) => {
+
+            var termianlNames = Object.keys(this.terminalList);
```
Comment (zikalino) [inline src/sshRunner.ts]: misspelled variable name

File: src/sshRunner.ts [hunk 2]
```diff
@@ -99,48 +111,78 @@ export class SSHRunner extends TerminalBaseRunner {
...
-
-                            if (utilities.isTelemetryEnabled()) {
-                                terminal.sendText('export ' + Constants.UserAgentName + '=' + utilities.getUserAgent());
+            }
+        }
+
+        if (terminal) {
+            terminal.show();
+
+            cmds.push('ansible-playbook ' + targetPlaybook);
```
Comment (zikalino) [inline src/sshRunner.ts]: How will this work with other PR which introduced ansible-playbook params?

File: src/sshRunner.ts [hunk 3]
```diff
@@ -99,48 +111,78 @@ export class SSHRunner extends TerminalBaseRunner {
...
-                    }
-                }, 500);
-            });
+    private getTerminalName(host: string): string {
+        return 'SSH ' + host;
+    }
 
+    private sendCommandsToTerminal(terminal: vscode.Terminal, cmds: string[]): void {
+        if (utilities.isTelemetryEnabled()) {
+            terminal.sendText('export ' + Constants.UserAgentName + '=' + utilities.getUserAgent());
```
Comment (zikalino) [inline src/sshRunner.ts]: so this will be sent to terminal even if terminal is reused and these variables are already set?

Comment (zikalino) [reply src/sshRunner.ts]: actually i think there should be if (telemetry) sendCommands... (telemetry commands) when terminal is created.

File: src/sshRunner.ts (hunk 2 above)
Comment (yungezz) [reply src/sshRunner.ts]: merge. so  pls review them, then i can do merge

File: src/sshRunner.ts (hunk 3 above)
Comment (yungezz) [reply src/sshRunner.ts]: those variables may change.

**Comment by yungezz** [comment]
telemetry configuration may change. that's why it's sent everytime. actually changed envs are one of concern  that I hesitate to add reuse terminal, as we discussed before.

**Pull request merged**